backend/dal/Configuration/AddressConfiguration.cs

```diff
@@ -26,14 +26,13 @@ public override void Configure(EntityTypeBuilder<Address> builder)
             builder.Property(m => m.ProvinceId).HasMaxLength(2);
             builder.Property(m => m.ProvinceId).IsRequired();
 
-            builder.HasOne(m => m.City).WithMany().HasForeignKey(m => m.CityId).OnDelete(DeleteBehavior.ClientSetNull);
             builder.HasOne(m => m.Province).WithMany().HasForeignKey(m => m.ProvinceId).OnDelete(DeleteBehavior.ClientSetNull);
 
-            builder.HasIndex(m => new { m.Id, m.CityId }).IncludeProperties(m => new { m.Address1, m.Address2 });
+            builder.HasIndex(m => new { m.Id, m.AdministrativeArea }).IncludeProperties(m => new { m.Address1, m.Address2 });
```

Comment:
AdministrativeArea is just a string.  Theoretically the address can have any value, but we will control the value by validating it against a list.